app/components/repository-migration-modal.js

```diff
@@ -0,0 +1,12 @@
+import Component from '@ember/component';
+import { action } from 'ember-decorators/object';
+
+export default Component.extend({
+  classNames: ['repository-migration-modal'],
+
+  @action
+  startMigration() {
+    this.get('repository').startMigration();
+    return this.get('onClose')();
```

Comment:
TL;DR: I'm not sure if adding no-ops makes things any more clear.

First, there's no connection between defining a property on a component and what arguments it can be passed. I can define `repository: null` on the component and that could mean it's a default value for state management used by a CP somewhere else within the component or that it's a placeholder to be overwritten by a passed-in value; it's quite ambiguous. I'd guess that 99% of the time, a component is representing a section of UI and tied to the DOM via its template, so you'd find its public API in how it's invoked (or, in the case of an addon, in its documentation), which is inevitably done in the template. If we wanted to be strict, we'd have to define something that checked passed params, which would involve overwriting a different framework hook. So I'm not sure this helps in that regard.

Second, in the case of `onClose()`, if we define the no-op and don't pass our own `onClose` action in the template, the no-op is invoked (instead of receiving an error that it doesn't exist). I find the error message more explicit about what is actually required, as users are forced to fix the exception to proceed.

@drogus @artursmirnov That said, if both of you are of a different opinion, I'm happy to make the change.

Reply:
@clekstro thank you for clarification! It totally makes sense. That was just an opinionated suggestion from me :)